### PR TITLE
Bg_Numa_Hana (Wooden Flower in Woodfall Temple) OK and documented, partial documentation of object_numa_obj

### PR DESCRIPTION
**Pull request by tom-overton**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

Pretty simple actor; I picked it up in case it interacted with En_Raf in any way, but it looks like it does not. Also documented the wooden flower pieces of object_num_obj.

### CODE REVIEW

**Review by AngheloAlf — APPROVED** [review]

File: src/overlays/actors/ovl_Bg_Numa_Hana/z_bg_numa_hana.h [hunk 1]
```diff
@@ -3,15 +3,41 @@
...
+} BgNumaHanaType;
+
 struct BgNumaHana;
 
 typedef void (*BgNumaHanaActionFunc)(struct BgNumaHana*, GlobalContext*);
 
+typedef struct {
+    /* 0x00 */ Vec3f pos;
+    /* 0x0C */ Vec3s rot;
+    /* 0x12 */ char unk_12[0x2];
```
Comment (hensldm) [inline src/overlays/actors/ovl_Bg_Numa_Hana/z_bg_numa_hana.h:21]:
```suggestion
    /* 0x12 */ UNK_TYPE1 unk_12[0x2];
```

**Review by hensldm — APPROVED** [review]
One minor comment

File: src/overlays/actors/ovl_Bg_Numa_Hana/z_bg_numa_hana.h (hunk 1 above)
Comment (tom-overton) [reply src/overlays/actors/ovl_Bg_Numa_Hana/z_bg_numa_hana.h:21]: I was almost certain I did this, but apparently not. Thanks for catching it.

File: src/overlays/actors/ovl_Bg_Numa_Hana/z_bg_numa_hana.c [hunk 2]
```diff
@@ -35,63 +42,380 @@ const ActorInit Bg_Numa_Hana_InitVars = {
...
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Numa_Hana/func_80A1AA28.s")
+void BgNumaHana_Update(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    BgNumaHana* this = THIS;
+    s32 type;
+    Vec3f firePos;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Numa_Hana/func_80A1AA38.s")
+    type = BG_NUMA_HANA_GET_TYPE(&this->dyna.actor);
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_Bg_Numa_Hana/z_bg_numa_hana.c:358]: Can be assigned where declared.

**Review by EllipticEllipsis — COMMENTED** [review]
Nice job on this, couple of minor things.

**Comment by EllipticEllipsis** [comment]
Jenkins seems to be mad because this doesn't match?

**Comment by tom-overton** [comment]
Yeah I accidentally did `UNK_TYPE` instead of `UNK_TYPE1` so the struct was way too big, oops.

**Review by EllipticEllipsis — APPROVED** [review]

**Review by Kenix3 — APPROVED** [review]
Good work on the documentation!

**Pull request merged**